File: src/main/org/nlogo/prim/etc/_observercode.java
package org.nlogo.prim.etc;

import org.nlogo.nvm.Command;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Syntax;

public final strictfp class _observercode
    extends Command
    implements org.nlogo.nvm.CustomAssembled {
  @Override
  public void perform(final Context context) {
    throw new UnsupportedOperationException();
  }

  @Override
  public Syntax syntax() {
    return Syntax.commandSyntax("O---", false);
  }

  public void assemble(org.nlogo.nvm.AssemblerAssistant a) {
    // do nothing -- drop out of existence
  }
}

File: src/main/org/nlogo/prim/etc/_observercode.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.api.Syntax
import org.nlogo.nvm.{ AssemblerAssistant, Command, Context, CustomAssembled }

class _observercode extends Command with CustomAssembled {
  override def syntax =
    Syntax.commandSyntax("O---", false)
  override def perform(context: Context) {
    throw new UnsupportedOperationException
  }
  override def assemble(a: AssemblerAssistant) {
    // by doing nothing here, drop out of existence
  }
}
